core/bentley/src/BentleyError.ts

```diff
@@ -658,3 +658,60 @@ export class BentleyError extends Error {
     }
   }
 }
+
+function hasProperty<T extends string>(obj: object, propertyName: T): obj is { [key in T]: unknown } {
+  return propertyName in obj;
+}
+
+/** @internal */
+export function getErrorStack(error: unknown): string | undefined {
+  if (typeof error === "object" && error !== null && hasProperty(error, "stack") && typeof error.stack === "string")
+    return error.stack;
+
+  return undefined;
+}
+
+/** @internal */
+export function getErrorMessage(error: unknown): string | undefined {
```

Comment:
Why is this is @internal, won't everyone need this? A little documentation would help. Maybe it should always return a string (or have another method that does) so everyone doesn't have to do `?? "unknown error"?

Reply:
or better yet, just return an empty string. I don't think "unknown error" helps anything.

Reply:
Good point, I'll make these public and document them...

Also, I like the idea of returning empty string.  That way, you could still do something like `|| "something useful"` if you want...

Reply:
Just pushed those changes.